src/features/customerList/customerListTableFilters/CustomerListTableFiltersForm.tsx

```diff
@@ -0,0 +1,426 @@
+          id="winterStorageAreaIds"
+          name="winterStorageAreaIds"
+          label={t('customerList.filters.winterStorageAreaIds')}
+          options={winterStorageAreaOptions}
+          placeholder={!isSectionEnabled.unmarkedWinterStorageSection ? undefined : t('common.all')}
+          value={winterStorageAreaOptions?.filter((option) => winterStorageAreaIds?.includes(option.value))}
+          onChange={onFieldChange}
+        />
+      </div>
+      <div className={styles.actionRow}>
+        <div className={styles.controlsContainer}>
+          <Checkbox
+            id="dateSection"
+            name="dateSection"
+            label={`${t('customerList.filters.dateInterval')}:`}
+            checked={isSectionEnabled.dateSection}
+            onChange={handleSectionChange}
+          />
+          <DateInput
+            disabled={!isSectionEnabled.dateSection}
+            id="startDate"
+            name="startDate"
+            value={startDate}
+            onChange={onFieldChange}
+          />
+          <DateInput
+            disabled={!isSectionEnabled.dateSection}
+            id="endDate"
+            name="endDate"
+            value={endDate}
+            onChange={onFieldChange}
+          />
+          <Checkbox
+            id="moreThanOneBerthOrWinterStorage"
+            name="moreThanOneBerthOrWinterStorage"
+            label={t('customerList.filters.moreThanOneBerthOrWinterStorage')}
+            checked={moreThanOneBerthOrWinterStorage}
+            onChange={onFieldChange}
+          />
+        </div>
+        <div className={styles.buttonContainer}>
+          <Button variant="secondary" onClick={handleReset}>
+            {t('customerList.message.resetFilters')}
+          </Button>
+          <Button type="submit">{t('customerList.message.useFilters')}</Button>
+        </div>
+      </div>
+    </form>
+  );
+};
+
+type MultiSelectProps = Omit<
+  ComboboxProps<Option>,
+  | 'defaultValue'
+  | 'options'
+  | 'value'
+  | 'onChange'
+  | 'toggleButtonAriaLabel'
+  | 'clearButtonAriaLabel'
+  | 'selectedItemRemoveButtonAriaLabel'
+> & {
+  id: string;
+  name: keyof ControlledFilters;
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  value?: Option[];
+  options?: Option[];
+};
+
+const MultiSelect = ({ name, onChange, value, options, disabled, placeholder, ...rest }: MultiSelectProps) => {
+  const { t } = useTranslation();
+
+  const handleChange = (options: Option[]) => {
+    const fakeChangeEvent = ({
+      target: {
+        name,
+        value: options.map((option: Option) => option.value),
+      },
+    } as unknown) as React.ChangeEvent<HTMLInputElement>;
+
+    onChange(fakeChangeEvent);
+  };
+
+  return (
+    <Combobox<Option>
+      {...rest}
+      multiselect={true}
+      disabled={disabled || !options}
+      placeholder={!disabled && !options ? t('customerList.message.loadingOptions') : placeholder}
```

Comment:
Can be confusing as to why `!disabled && !options` means that options are loading. isLoading prop or something like that would make the loading logic little bit more understandable.

Reply:
Now reader of the code has to read this component to understand when input is in loading state